libs.lsp/nbproject/project.properties

```diff
@@ -0,0 +1,27 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+javac.source=1.7
+javac.compilerargs=-Xlint -Xlint:-serial
+release.external/gson-2.7.jar=modules/ext/gson-2.7.jar
+release.external/guava-21.0.jar=modules/ext/guava-21.0.jar
+release.external/org.eclipse.lsp4j-0.4.1.jar=modules/ext/org.eclipse.lsp4j-0.4.1.jar
+release.external/org.eclipse.lsp4j.generator-0.4.1.jar=modules/ext/org.eclipse.lsp4j.generator-0.4.1.jar
+release.external/org.eclipse.lsp4j.jsonrpc-0.4.1.jar=modules/ext/org.eclipse.lsp4j.jsonrpc-0.4.1.jar
```

Comment:
I assume these libraries are themselves OSGi bundles. If so, I prefer to package them and consume them like OSGi bundles. Just like Mylyn does with its OSGi bundles.

Reply:
In the context of friend dependencies discussion (raging on in parallel), I see an attempt to bring private only copy of popular libraries like `gson` or `guava` going the exactly opposite way. Attitude like this brought NetBeans to the "overuse of friend deps" which leads to hacks and tricks to workaround it.

I prefer exposing popular 3rd party libraries (especially those packaged as OSGi bundles) as reusable components.

Reply:
I think there's a significant difference between e.g. html.editor and guava. It is not reasonably possible to have multiple copies of html.editor in the IDE, while the side-effects of having multiple copies of guava should be fairly limited. I.e. if some other module needs guava, there's no need to do "hacks and tricks", it can have its own copy.

I wonder if there may be multiple versions of an OSGi module under the NB-OSGi combined module system, and if it will select the proper one based on dependencies.

If not, then we may putting ourselves into a corner where upgrading various libraries will lead to significant amount of time spent ensuring things still work after the upgrade. And the time, I think, might be better spent on actual features and bugfixes of this project.